[Regression 2025-08-26] `- vzNAT: true` produces error `inspecting instance: [field networks[0].lima  or  field networks[0].socket must be set]`

### Description

With following entry in `lima.yaml`:
```yaml
networks:
- vzNAT: true
```
`limactl start` prints error:
```console
$ limactl start docker
WARN[0000] Both top-level 'rosetta' and 'vmOpts.vz.rosetta' are configured. Using vmOpts.vz.rosetta. Top-level 'rosetta' is deprecated. 
INFO[0000] Using the existing instance "docker"         
FATA[0000] errors inspecting instance: [field `networks[0].lima` or  field `networks[0].socket must be set] 
```

6aceec31e50369425f83258b2355acf0b9b6d876 causes this.
That also broke `linactl prune`, `limactl restart`:
```console
➜ ~14:28:10 14:28:21~ ((6aceec31)) limactl prune --keep-referred 
WARN[0000] Both top-level 'rosetta' and 'vmOpts.vz.rosetta' are configured. Using vmOpts.vz.rosetta. Top-level 'rosetta' is deprecated. 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x58 pc=0x103051cbc]

goroutine 1 [running]:
main.locationsFromLimaYAML(0x0)
	/Users/norio/ghq/github.com/lima-vm/lima/cmd/limactl/prune.go:110 +0x2c
main.knownLocations({0x103642cb0, 0x104648f60})
	/Users/norio/ghq/github.com/lima-vm/lima/cmd/limactl/prune.go:83 +0x90
main.pruneAction(0x140004f5b08?, {0x140004d8900?, 0x4?, 0x1030bd571?})
	/Users/norio/ghq/github.com/lima-vm/lima/cmd/limactl/prune.go:52 +0xbc
github.com/spf13/cobra.(*Command).execute(0x140004f5b08, {0x140004d88e0, 0x2, 0x2})
	/Users/norio/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1015 +0x7d4
github.com/spf13/cobra.(*Command).ExecuteC(0x140004f4008)
	/Users/norio/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1148 +0x350
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/norio/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1071
main.executeWithPluginSupport(0x140004f4008?, {0x14000142010?, 0x3?, 0x3?})
	/Users/norio/ghq/github.com/lima-vm/lima/cmd/limactl/main.go:227 +0xbc
main.main()
	/Users/norio/ghq/github.com/lima-vm/lima/cmd/limactl/main.go:46 +0x54
➜ ~14:28:21 14:28:58~ ((6aceec31)) limactl restart docker
WARN[0000] Both top-level 'rosetta' and 'vmOpts.vz.rosetta' are configured. Using vmOpts.vz.rosetta. Top-level 'rosetta' is deprecated. 
WARN[0000] The instance is not running, continuing with the restart 
WARN[0000] Both top-level 'rosetta' and 'vmOpts.vz.rosetta' are configured. Using vmOpts.vz.rosetta. Top-level 'rosetta' is deprecated. 
INFO[0000] Starting the instance "docker" with  VM driver "" 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x298 pc=0x1051ba650]

goroutine 1 [running]:
github.com/lima-vm/lima/v2/pkg/instance.Prepare({0x1058cecb0, 0x1068d4f60}, 0x14000568140)
	/Users/norio/ghq/github.com/lima-vm/lima/pkg/instance/start.go:84 +0x30
github.com/lima-vm/lima/v2/pkg/instance.Start({0x1058cecb0, 0x1068d4f60}, 0x14000568140, {0x0, 0x0}, 0x0, 0x0)
	/Users/norio/ghq/github.com/lima-vm/lima/pkg/instance/start.go:191 +0x1ac
github.com/lima-vm/lima/v2/pkg/instance.Restart({0x1058cecb0, 0x1068d4f60}, 0x14000568140)
	/Users/norio/ghq/github.com/lima-vm/lima/pkg/instance/restart.go:29 +0x68
main.restartAction(0x1400058db08, {0x1400054e960?, 0x4?, 0x105349571?})
	/Users/norio/ghq/github.com/lima-vm/lima/cmd/limactl/restart.go:48 +0xf0
github.com/spf13/cobra.(*Command).execute(0x1400058db08, {0x1400054e940, 0x2, 0x2})
	/Users/norio/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1015 +0x7d4
github.com/spf13/cobra.(*Command).ExecuteC(0x1400056c008)
	/Users/norio/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1148 +0x350
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/norio/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1071
main.executeWithPluginSupport(0x1400056c008?, {0x140001ba010?, 0x3?, 0x3?})
	/Users/norio/ghq/github.com/lima-vm/lima/cmd/limactl/main.go:227 +0xbc
main.main()
	/Users/norio/ghq/github.com/lima-vm/lima/cmd/limactl/main.go:46 +0x54
```

--- a/pkg/driver/qemu/qemu_driver.go
+++ b/pkg/driver/qemu/qemu_driver.go
@@ -36,6 +36,7 @@ import (
 	"github.com/lima-vm/lima/v2/pkg/networks/usernet"
 	"github.com/lima-vm/lima/v2/pkg/osutil"
 	"github.com/lima-vm/lima/v2/pkg/ptr"
+	"github.com/lima-vm/lima/v2/pkg/reflectutil"
 	"github.com/lima-vm/lima/v2/pkg/version/versionutil"
 )
 
@@ -95,6 +96,18 @@ func validateConfig(ctx context.Context, cfg *limatype.LimaYAML) error {
 	}
 	if err := validateMountType(cfg); err != nil {
 		return err
+	}
+
+	for i, nw := range cfg.Networks {
+		if unknown := reflectutil.UnknownNonEmptyFields(nw,
+			"Lima",
+			"Socket",
+			"MACAddress",
+			"Metric",
+			"Interface",
+		); len(unknown) > 0 {
+			logrus.Warnf("vmType %s: ignoring networks[%d]: %+v", *cfg.VMType, i, unknown)
+		}
 	}
 
 	if cfg.VMOpts.QEMU.MinimumVersion != nil {

--- a/pkg/driver/vz/vz_driver_darwin.go
+++ b/pkg/driver/vz/vz_driver_darwin.go
@@ -238,25 +238,6 @@ func validateConfig(_ context.Context, cfg *limatype.LimaYAML) error {
 		); len(unknown) > 0 {
 			logrus.Warnf("vmType %s: ignoring networks[%d]: %+v", *cfg.VMType, i, unknown)
 		}
-
-		field := fmt.Sprintf("networks[%d]", i)
-		switch {
-		case nw.Lima != "":
-			if nw.VZNAT != nil && *nw.VZNAT {
-				return fmt.Errorf("field `%s.lima` and field `%s.vzNAT` are mutually exclusive", field, field)
-			}
-		case nw.Socket != "":
-			if nw.VZNAT != nil && *nw.VZNAT {
-				return fmt.Errorf("field `%s.socket` and field `%s.vzNAT` are mutually exclusive", field, field)
-			}
-		case nw.VZNAT != nil && *nw.VZNAT:
-			if nw.Lima != "" {
-				return fmt.Errorf("field `%s.vzNAT` and field `%s.lima` are mutually exclusive", field, field)
-			}
-			if nw.Socket != "" {
-				return fmt.Errorf("field `%s.vzNAT` and field `%s.socket` are mutually exclusive", field, field)
-			}
-		}
 	}
 
 	switch audioDevice := *cfg.Audio.Device; audioDevice {

--- a/pkg/limayaml/validate.go
+++ b/pkg/limayaml/validate.go
@@ -449,11 +449,24 @@ func validateNetwork(y *limatype.LimaYAML) error {
 			if nw.Socket != "" {
 				errs = errors.Join(errs, fmt.Errorf("field `%s.lima` and field `%s.socket` are mutually exclusive", field, field))
 			}
+			if nw.VZNAT != nil && *nw.VZNAT {
+				errs = errors.Join(errs, fmt.Errorf("field `%s.lima` and field `%s.vzNAT` are mutually exclusive", field, field))
+			}
 		case nw.Socket != "":
+			if nw.VZNAT != nil && *nw.VZNAT {
+				errs = errors.Join(errs, fmt.Errorf("field `%s.socket` and field `%s.vzNAT` are mutually exclusive", field, field))
+			}
 			if fi, err := os.Stat(nw.Socket); err != nil && !errors.Is(err, os.ErrNotExist) {
 				errs = errors.Join(errs, err)
 			} else if err == nil && fi.Mode()&os.ModeSocket == 0 {
 				errs = errors.Join(errs, fmt.Errorf("field `%s.socket` %q points to a non-socket file", field, nw.Socket))
+			}
+		case nw.VZNAT != nil && *nw.VZNAT:
+			if nw.Lima != "" {
+				errs = errors.Join(errs, fmt.Errorf("field `%s.vzNAT` and field `%s.lima` are mutually exclusive", field, field))
+			}
+			if nw.Socket != "" {
+				errs = errors.Join(errs, fmt.Errorf("field `%s.vzNAT` and field `%s.socket` are mutually exclusive", field, field))
 			}
 		default:
 			errs = errors.Join(errs, fmt.Errorf("field `%s.lima` or  field `%s.socket must be set", field, field))

